sideboard/internal/connection_checker.py

```diff
@@ -18,35 +19,41 @@ def _check(url, **ssl_params):
         host = parsed.hostname
         port = parsed.port or (443 if parsed.scheme in ['https', 'wss'] else 80)
         status.append('using hostname {} and port {}'.format(host, port))
-    
+
     try:
         ip = socket.gethostbyname(host)
     except Exception as e:
         return status + ['failed to resolve host with DNS: {!s}'.format(e)]
     else:
         status.append('successfully resolved host {} to {}'.format(host, ip))
-    
+
     sock = None
     try:
         sock = socket.create_connection((host, port))
     except Exception as e:
         return status + ['failed to establish a socket connection to {} on port {}: {!s}'.format(host, port, e)]
     else:
         status.append('successfully opened socket connection to {}:{}'.format(host, port))
-    
+
     if any(ssl_params.values()):
         try:
             wrapped = ssl.wrap_socket(sock, **ssl_params)
         except Exception as e:
             return status + ['failed to complete SSL handshake ({}): {!s}'.format(ssl_params, e)]
         else:
-            status.append('succeeded at SSL handshake')
+            status.append('succeeded at SSL handshake (without validating server cert)')
         finally:
             if sock:
                 sock.close()
-    
+
+        try:
+            with closing(socket.create_connection((host, port))) as sock:
+                wrapped = ssl.wrap_socket(sock, cert_reqs=ssl.CERT_REQUIRED, **ssl_params)
+        except Exception as e:
+            return status + ['failed to validate server cert ({}): {!s}'.format(ssl_params, e)]
+
```

Comment:
This adds a single extra step/check to the set of checks explained in my rant above.